util/network/src/lib.rs

```diff
@@ -145,14 +148,61 @@ impl NonReservedPeerMode {
 	}
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "ipc", binary)]
+pub struct IpFilter {
+    pub predefined: AllowIP,
+    pub custom_allow: Vec<IpNetwork>,
+    pub custom_block: Vec<IpNetwork>,
+} 
+
+impl IpFilter {
+    pub fn new() -> IpFilter {
+        IpFilter {
+            predefined: AllowIP::All,
+            custom_allow: vec![],
+            custom_block: vec![],
+        }
+    }
+
+    /// Attempt to parse the peer mode from a string.
+    pub fn parse(s: &str) -> Result<IpFilter, IpNetworkError> {
+        let mut filter = IpFilter::new();
+        for f in s.split_whitespace() {
+            match f {
+                "all" => filter.predefined = AllowIP::All,
+                "private" => filter.predefined = AllowIP::Private,
+                "public" => filter.predefined = AllowIP::Public,
+                "none" => filter.predefined = AllowIP::Non,
+                custom => {
+                    if custom.starts_with("-") {
+                        match IpNetwork::from_str(&custom.to_owned().split_off(1)) {
```

Comment:
Could be just `filter.custom_block.push(IpNetwork::from_str(&custom.to_owned().split_off(1)?)`; Same for the `match` below.